dagobah/core/job.py

```diff
@@ -467,12 +469,119 @@ def initialize_snapshot(self):
         if not is_valid:
             raise DagobahError(reason)
 
-        self.snapshot = snapshot_to_validate
+        verified = self.verify()
+
+        if not verified:
+            raise DagobahError("Job has a cycle, possibly within another Job "
+                               + "reference")
+
+        # Due to thread locks in underlying tasks, they must be manually copied
+        tasks_copy = {}
+        for name, task in self.tasks.iteritems():
+            if self.implements_runnable(task):
+                tasks_copy[name] = Task(self, task.command, task.name,
+                                        soft_timeout=task.soft_timeout,
+                                        hard_timeout=task.hard_timeout,
+                                        hostname=task.hostname)
+            elif self.implements_expandable(task):
+                tasks_copy[name] = JobTask(self, task.name,
+                                           task.target_job_name)
+            else:
+                raise DagobahError("Malformed task neither a Task nor JobTask")
```

Comment:
Sure, they're functionally equivalent. If it's an instance method you might want to call it `.clone()`